src/components/movie-list-page/MovieListPage.jsx

```diff
@@ -0,0 +1,193 @@
+import React, { useEffect, useState } from "react";
+import { useSearchParams, Outlet, useNavigate } from "react-router-dom";
+import GenreSelect from "../genre-select/GenreSelect";
+import MovieTile from "../movie-tile/MovieTile";
+import SortControl from "../sort-control/SortControl";
+import Dialog from "../dialog/Dialog";
+import MovieForm from "../movie-form/MovieForm";
+import "./MovieListPage.css";
+
+const deleteMessage = (
+  <div className="delete-message">
+    <p>Are you sure you want to delete this movie?</p>
+    <button>CONFIRM</button>
+  </div>
+);
+
+export default function MovieListPage() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [selectedMovie, setSelectedMovie] = useState(null);
+  const [movieList, setMovieList] = useState([]);
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [editedMovie, setEditedMovie] = useState({});
+  const [dialogTitle, setDialogTitle] = useState("");
+
+  const sortCriterion = searchParams.get("sort") || "release_date";
+  const searchQuery = searchParams.get("search") || "";
+  const activeGenre = searchParams.get("genre") || "ALL";
+
+  const navigate = useNavigate();
+
+  const url = `http://localhost:4000/movies?limit=100&search=${searchQuery}&filter=${
+    activeGenre === "ALL" ? "" : activeGenre
+  }&searchBy=title&sortBy=${sortCriterion}&sortOrder=asc`;
+
+  useEffect(() => {
```

Comment:
as an idea for refactoring: we can encapsulate this logic in a custom hook which uses `useSearchParams`, `useEffect` and `useState` hooks and returns the data we need (e.g.,  `movieList`, `searchQuery`, `activeGenre`, `sortCriterion`, etc)